stdlib/public/core/UTF16.swift

```diff
@@ -434,3 +434,430 @@ extension Unicode.UTF16.ForwardParser: Unicode.Parser, _UTFParser {
     return r
   }
 }
+
+private enum ScalarFallbackResult: UInt8 {
+  case invalid
+  case singleByte
+  case multiByte
+}
+
+#if arch(arm64_32)
+typealias Word = UInt64
+#else
+typealias Word = UInt
+#endif
+//Blocks containing only ASCII will not have any of these bits set
+@_transparent var asciiMask:Word {
+  Word(truncatingIfNeeded: 0xFF80FF80_FF80FF80 as UInt64)
+}
+
+/*
+ Shifting a UInt16 value by 7 bits gets us:
+ <0x80: 0 (ASCII, 1 byte UTF8)
+ 0x80-0x7FF: 1-15 (2 byte UTF8)
+ 0x800-0x3FFF: 16-127 (3 byte UTF8)
+ Which we can use as an index into this table.
+ Values >= 0x4000 are handled via the scalar path to keep the table size down
+ */
+@_transparent func withLengthLUT(
+  _ work: (UnsafeRawBufferPointer) -> Int
+) -> Int {
+  let lut: (
```

Comment:
We can probably use an `InlineArray<128, UInt8>` here.

Reply:
Oh I didn't read your latest message that said you came from an InlineArray. Any reason you switched away from it?

Reply:
Availability issues :/